tf_quant_finance/black_scholes/vanilla_prices.py

```diff
@@ -182,6 +182,262 @@ def option_price(*,
+      `cost_of_carries` must be supplied.
+      Default value: `None`, calculated from `cost_of_carries`.
+    cost_of_carries: A optional real `Tensor` of same dtype as the
+      `volatilities` and of the shape that broadcasts with `volatilities`.
+      Cost of storing a physical commodity, the cost of interest paid when
+      long, or the opportunity cost, o the cost of paying dividends when short.
+      If not `None`, `continuous_dividends` is calculated as r - c,
+      where r are the `discount_rates` and c is `cost_of_carries`.
+    barriers: A real `Tensor` of same dtype as the `volatilities` and of the
+      shape that broadcasts with `volatilities`. The barriers of each option.
+    rebates: A real `Tensor` of same dtype as the `volatilities` and of the
+      shape that broadcasts with `volatilities`. A rebates contingent upon
+      reaching the barriers price.
+    is_barrier_down: A real `Tensor` of `boolean` values and of the shape
+      that broadcasts with `volatilities`. True if barrier is below spot
+      else false.
+    is_knock_out: A real `Tensor` of `boolean` values and of the shape
+      that broadcasts with `volatilities`. True if option is knock out
+      else false.
+    is_call_options: A real `Tensor` of `boolean` values and of the shape
+      that broadcasts with `volatilities`. True if option is call else
+      false.
+    dtype: Optional `tf.DType`. If supplied, the dtype to be used for conversion
+      of any supplied non-`Tensor` arguments to `Tensor`.
+      Default value: `None` which maps to the default dtype inferred by
+      TensorFlow.
+    name: str. The name for the ops created by this function.
+      Default value: `None` which is mapped to the default name `option_price`.
+  Returns:
+    option_prices: A `Tensor` of same shape as `spots`. The approximate price of
+    the barriers option under black scholes.
+
+  """
+  if (continuous_dividends is None) == (cost_of_carries is None):
+    raise ValueError('At most one of continuous_dividends and cost of carries '
+                     'may be supplied')
+  with tf.name_scope(name or "barrier_price"):
+    # Convert all to tensor and enforce float dtype where required
+    if discount_rates is not None:
+      discount_rates = tf.convert_to_tensor(
+          discount_rates, dtype=dtype, name="discount_rates")
+    else:
+      discount_rates = tf.convert_to_tensor(
+          1, dtype=dtype, name="discount_rates")
+
+    dtype = discount_rates.dtype
+
+    if continuous_dividends is not None:
+      continuous_dividends = tf.convert_to_tensor(
+          continuous_dividends, dtype=dtype, name="continuous_dividends")
+
+    if cost_of_carries is not None:
+      continuous_dividends = tf.convert_to_tensor(
+          discount_rates-cost_of_carries, dtype=dtype,
+          name="continuous_dividends")
+
+    spots = tf.convert_to_tensor(
+        spots, dtype=dtype, name="spots")
```

Comment:
Please move this to the top and 
set 
`dtype = dtype or spots.dtype`

This is similar pattern elsewhere. See, e.g., [here](https://github.com/google/tf-quant-finance/blob/1239ce3a22b4a6545d1600835333e8c9984ade7d/tf_quant_finance/black_scholes/vanilla_prices.py#L136)

Reply:
Done!